recipes/_deps_centos.rb

```diff
@@ -0,0 +1,112 @@
+#
+# Cookbook Name:: mcrouter
+# Recipe:: _build_deps.centos.rb
+#
+# Copyright 2015 EverTrue, Inc.
+# Copyright 2017 Roblox, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+include_recipe 'build-essential'
+include_recipe 'yum'
+include_recipe 'yum-epel'
+
+cache_dir = Chef::Config[:file_cache_path]
+
+%w(
+  centos-release-scl
+  gcc
+  gcc-c++
+  devtoolset-4-gcc
+  devtoolset-4-gcc-c++
+  autoconf
+  binutils-devel
+  bison
+  boost
+  boost-devel
+  double-conversion-devel
+  flex
+  git
+  glog-devel
+  jemalloc-devel
+  libtool
+  libevent-devel
+  make
+  openssl-devel
+  python-devel
+  scons
+  ragel
+  ruby
+  ruby-devel
+  cyrus-sasl
+  cyrus-sasl-devel
+  python-libs
+  python-devel
+  numactl-libs
+  numactl-devel
+  snappy
+  snappy-devel
+  gflags
+  gflags-devel
+  glog
+  automake
+  libevent
+  libatomic
+  rpm-build
+  bzip2-devel
+  libcap-devel
+  unzip
+  glibc-devel.i686
+  glibc-devel.x86_64
+  zlib-devel
+  thrift
+  thrift-devel
+  double-conversion-devel
+).each do |pkg|
+  package pkg
+end
+
+remote_file "#{cache_dir}/libarchive-3.1.2-10.el7_2.x86_64.rpm" do
```

Comment:
Also, why a specific libarchive version? It seems like there are a lot of custom package installs in this recipe that are going to be very brittle.